hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineStateManagerV2Impl.java

```diff
@@ -246,10 +246,10 @@ public void updatePipelineState(
       HddsProtos.PipelineID pipelineIDProto, HddsProtos.PipelineState newState)
       throws IOException {
     PipelineID pipelineID = PipelineID.getFromProtobuf(pipelineIDProto);
-    Pipeline.PipelineState oldState =
-        getPipeline(pipelineID).getPipelineState();
-    lock.writeLock().lock();
+    Pipeline.PipelineState oldState = null;
     try {
+      oldState = getPipeline(pipelineID).getPipelineState();
```

Comment:
Do we need to call getPipeline also under lock?

Reply:
It should be fine as is , as it ignores the PipelineNotFoundException.